agent/recorder-agent.c

```diff
@@ -124,6 +134,15 @@ hwangsae_recorder_agent_send_rest_api (HwangsaeRecorderAgent * self,
   return;
 }
 
+static void
+stream_disconnected_cb (HwangsaeRecorder * recorder,
+    HwangsaeRecorderAgent * self)
+{
+  g_debug ("Stream disconnected, cleaning recorder");
+
+  g_clear_object (&recorder);
```

Comment:
I'm not sure if this is the correct way of cleaning the HwangsaeRecorder. I understand that I need to use a signal as I need to be sure that the pipeline has been cleaned, but maybe I'm missing something.